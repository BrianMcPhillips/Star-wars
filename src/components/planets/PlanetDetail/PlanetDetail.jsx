import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const PlanetDetail = ({
  name,
  diameter,
  climate,
  gravity,
  terrain,
  population,
  residents,
  films
}) => (
  <div>
    <h1>{name}</h1>
    <h3>Diameter: {diameter}</h3>
    <h4>Climate: {climate}</h4>
    <h4>Gravity: {gravity}</h4>
    <h4>Terrain: {terrain}</h4>
    <h4>Population: {population}</h4>
    <h4>Residents: {residents.map((resident, i) => 
      <Link
        to={`/char-detail/${resident.split('people/')[1]}`}
        key={resident + i}>
        <p>{resident}</p>
      </Link>
    )}</h4>
    <h4>Films: {films.map((film, i) => 
      <Link
        to={`/film-detail/${film.split('films/')[1]}`}
        key={film + i}>
        <p>{film}</p>
      </Link>
    )}</h4>
  </div>
);

PlanetDetail.propTypes = {
  name: PropTypes.string.isRequired,
  diameter: PropTypes.string.isRequired,
  climate: PropTypes.string.isRequired,
  gravity: PropTypes.string.isRequired,
  terrain: PropTypes.string.isRequired,
  population: PropTypes.string.isRequired,
  residents: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  films: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default PlanetDetail;
