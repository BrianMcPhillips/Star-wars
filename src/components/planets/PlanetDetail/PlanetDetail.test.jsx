import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PlanetDetail from './PlanetDetail';


describe('PlanetDetail component', () => {
  afterEach(() => cleanup());
  it('renders PlanetDetail', () => {
    const { asFragment } = render(
      <PlanetDetail 
        name="Tatooine"
        diameter="10465"
        climate="arid"
        gravity= "1 standard"
        terrain="desert"
        population= "200000"
        residents= {[
          'https://swapi.dev/api/people/1/',
          'https://swapi.dev/api/people/2/',
          'https://swapi.dev/api/people/4/',
          'https://swapi.dev/api/people/6/',
          'https://swapi.dev/api/people/7/',
          'https://swapi.dev/api/people/8/',
          'https://swapi.dev/api/people/9/',
          'https://swapi.dev/api/people/11/',
          'https://swapi.dev/api/people/43/',
          'https://swapi.dev/api/people/62/'
        ]}
        films={[
          'https://swapi.dev/api/films/1/',
          'https://swapi.dev/api/films/3/',
          'https://swapi.dev/api/films/4/',
          'https://swapi.dev/api/films/5/',
          'https://swapi.dev/api/films/6/'
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
