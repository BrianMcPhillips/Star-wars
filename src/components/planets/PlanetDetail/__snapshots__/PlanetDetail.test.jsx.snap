// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetDetail component renders PlanetDetail 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Tatooine
    </h1>
    <h3>
      Diameter: 10465
    </h3>
    <h4>
      Climate: arid
    </h4>
    <h4>
      Gravity: 1 standard
    </h4>
    <h4>
      Terrain: desert
    </h4>
    <h4>
      Population: 200000
    </h4>
    <h4>
      Residents: 
      <a
        href="/char-detail/1/"
      >
        <p>
          https://swapi.dev/api/people/1/
        </p>
      </a>
      <a
        href="/char-detail/2/"
      >
        <p>
          https://swapi.dev/api/people/2/
        </p>
      </a>
      <a
        href="/char-detail/4/"
      >
        <p>
          https://swapi.dev/api/people/4/
        </p>
      </a>
      <a
        href="/char-detail/6/"
      >
        <p>
          https://swapi.dev/api/people/6/
        </p>
      </a>
      <a
        href="/char-detail/7/"
      >
        <p>
          https://swapi.dev/api/people/7/
        </p>
      </a>
      <a
        href="/char-detail/8/"
      >
        <p>
          https://swapi.dev/api/people/8/
        </p>
      </a>
      <a
        href="/char-detail/9/"
      >
        <p>
          https://swapi.dev/api/people/9/
        </p>
      </a>
      <a
        href="/char-detail/11/"
      >
        <p>
          https://swapi.dev/api/people/11/
        </p>
      </a>
      <a
        href="/char-detail/43/"
      >
        <p>
          https://swapi.dev/api/people/43/
        </p>
      </a>
      <a
        href="/char-detail/62/"
      >
        <p>
          https://swapi.dev/api/people/62/
        </p>
      </a>
    </h4>
    <h4>
      Films: 
      <a
        href="/film-detail/1/"
      >
        <p>
          https://swapi.dev/api/films/1/
        </p>
      </a>
      <a
        href="/film-detail/3/"
      >
        <p>
          https://swapi.dev/api/films/3/
        </p>
      </a>
      <a
        href="/film-detail/4/"
      >
        <p>
          https://swapi.dev/api/films/4/
        </p>
      </a>
      <a
        href="/film-detail/5/"
      >
        <p>
          https://swapi.dev/api/films/5/
        </p>
      </a>
      <a
        href="/film-detail/6/"
      >
        <p>
          https://swapi.dev/api/films/6/
        </p>
      </a>
    </h4>
  </div>
</DocumentFragment>
`;
